addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,498 @@
+            values['credit_note_code'] = invoice.l10n_es_tbai_refund_reason or ('R5' if values['is_simplified'] else 'R1')
+            values['credit_note_invoice'] = invoice.reversed_entry_id
+
+        # === LINES (DETALLES) ===
+        values['invoice_lines'] = [{
+            'line': line,
+            'address': regex_sub(r'[^0-9a-zA-Z ]', '', line.name)[:250],
+        } for line in invoice.invoice_line_ids.filtered(lambda line: not line.display_type)]
+
+        # CODES (CLAVES): TODO there's 15 more codes to implement, also there can be up to 3 in total
+        # See https://www.gipuzkoa.eus/documents/2456431/13761128/Anexo+I.pdf/2ab0116c-25b4-f16a-440e-c299952d683d
+        com_partner = invoice.commercial_partner_id
+        if not com_partner.country_id or com_partner.country_id.code in eu_country_codes:
+            values['vat_regime_code'] = '01'
+        else:
+            values['vat_regime_code'] = '02'
+
+        # === BREAKDOWN TYPE (TIPO DESGLOSE) ===
+        edi_format_sii = self.env['account.edi.format']
+        if com_partner.country_id.code in ('ES', False) and not (com_partner.vat or '').startswith("ESN"):
+            tax_details_info_vals = edi_format_sii._l10n_es_edi_get_invoices_tax_details_info(invoice)
+            values['invoice_breakdown'] = tax_details_info_vals['tax_details_info']
+            values['invoice_total'] = round(-1 * (tax_details_info_vals['tax_details']['base_amount']
+                                                  + tax_details_info_vals['tax_details']['tax_amount']
+                                                  - tax_details_info_vals['tax_amount_retention']), 2)
+
+        else:
+            tax_details_info_service_vals = edi_format_sii._l10n_es_edi_get_invoices_tax_details_info(
+                invoice,
+                filter_invl_to_apply=lambda x: any(t.tax_scope == 'service' for t in x.tax_ids)
+            )
+            tax_details_info_consu_vals = edi_format_sii._l10n_es_edi_get_invoices_tax_details_info(
+                invoice,
+                filter_invl_to_apply=lambda x: any(t.tax_scope == 'consu' for t in x.tax_ids)
+            )
+
+            if tax_details_info_service_vals['tax_details_info']:
+                values['services_provision'] = tax_details_info_service_vals['tax_details_info']
+            if tax_details_info_consu_vals['tax_details_info']:
+                values['goods_delivery'] = tax_details_info_consu_vals['tax_details_info']
+
+            values['invoice_total'] = round(-1 * (
+                tax_details_info_service_vals['tax_details']['base_amount']
+                + tax_details_info_service_vals['tax_details']['tax_amount']
+                - tax_details_info_service_vals['tax_amount_retention']
+                + tax_details_info_consu_vals['tax_details']['base_amount']
+                + tax_details_info_consu_vals['tax_details']['tax_amount']
+                - tax_details_info_consu_vals['tax_amount_retention']), 2)
```

Comment:
create a sub-method for that in SII since all this stuff is copy-pasted from `_l10n_es_edi_get_invoices_info`

Reply:
TDB in other branch, see 
https://github.com/odoo/odoo/pull/87101